### PR TITLE
New cmd line utility "scli" (golang porting)

### LINKED ISSUE #39: Port scripts in bin/ to Golang
In high-scale scenarios (100+ clusters) some operations such as cluster import will benefit from parallelization.

There is no intention to implement that on the current Javascript code, therefore intention is to port those scripts to Golang.

### PR DESCRIPTION
**Pull request by fgiudici**
This PR includes the setup, teardown and get_access (now 'state') command porting in go.
There is one single cmd line binary (`scli`) that is built and put in the `bin` directory with:

```
$> make
```
run it with `bin/scli`:
```
$> bin/scli
NAME:
   scli - test Rancher at scale

USAGE:
   scli [global options] command [command options] 

COMMANDS:
   setup              Deploys the test environment
   state              Retrieves information of the deployed clusters
   teardown, destroy  Tears down the test environment (all the clusters)
   help, h            Shows a list of commands or help for one command

GLOBAL OPTIONS:
   --tf-dir value  terraform working directory (default: "terraform/main/k3d") [$TERRAFORM_WORK_DIR]
   --help, -h      show help

COPYRIGHT:
   (c) 2024 SUSE LLC

```
Related to #39

### CODE REVIEW

**Comment by moio** [comment]
Hey @fgiudici a lot quality work in here! Thanks a lot for your contribution!

Some general questions:

1. IIUC helm, kubectl and terraform are now integrated with/autoinstalled by the `scli` binary (awesome!). Is there an intention to integrate k6 as well?
2. I am not very familiar with the old golang vendoring approach - what is the reason to have the vendor directory contents checked in the repo?
3. What is the reason to vendor the Grafana chart instead of using the upstream release?

Keep up the good work 😉 
Thanks

File: pkg/kubectl/kubectl.go [hunk 1]
```diff
@@ -0,0 +1,304 @@
...
+						ConfigMap: &v1.ConfigMapVolumeSource{
+							LocalObjectReference: v1.LocalObjectReference{Name: "k6-lib-files"},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	podCli := cl.clientset.CoreV1().Pods(K6Namespace)
```
Comment (moio) [inline pkg/kubectl/kubectl.go:214]: What was the reason to prefer going via ther Kubernetes API here, instead of shelling out to kubectl?

Comment (fgiudici) [reply pkg/kubectl/kubectl.go:214]: client-go is the official library to access kubernetes and is the best (and hard) way.
Anyway, implementing the "wait" command is not trivial... so resorted out to Exec and kubectl binary.
All the kube api access is managed via the internal pkg/kubectl, so we may change the implementation of the functions there as we prefer and at any time without changing the "main" functions in the code.

**Comment by fgiudici** [comment]
Rebased and added a couple of options more

**Pull request merged**